src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -362,8 +362,11 @@ void QgsAllLayersFeaturesLocatorFilter::prepare( const QString &string, const Qg
     req.setSubsetOfAttributes( expression.referencedAttributeIndexes( layer->fields() ).toList() );
     if ( !expression.needsGeometry() )
       req.setFlags( QgsFeatureRequest::NoGeometry );
+    QString enhancedSearch = string;
+    enhancedSearch.replace( " ", "%" );
     req.setFilterExpression( QStringLiteral( "%1 ILIKE '%%2%'" )
-                             .arg( layer->displayExpression(), string ) );
+                             .arg( layer->displayExpression() )
```

Comment:
```suggestion
                             .arg( layer->displayExpression(), enhancedSearch ) );
```

should be more efficient